#### Практическое задание: Урок 6. Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.

In [1]:
import numpy as np

##### 1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. 

##### Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

Рассчитаем ковариацию 2 способами:

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [3]:
cov1 = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
cov1

9157.839999999997

In [4]:
cov2 = np.cov(zp,ks, ddof=0)[0][1]
cov2

9157.84

Рассчитаем среднее квадратичное отклонение для обоих массивов:

In [5]:
sigma_zp = np.std(zp)
sigma_ks = np.std(ks)
sigma_zp, sigma_ks

(59.115480206118605, 174.55340157098058)

Найдем коэффициент корреляции Пирсона (1 способ, 2 способ):

In [6]:
r_zp_ks1 = cov2 / (sigma_zp * sigma_ks)
r_zp_ks1

0.8874900920739162

In [7]:
r_zp_ks2 = np.corrcoef(zp, ks)[0, 1]
r_zp_ks2

0.8874900920739162

##### 2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Сигма не известна, поэтому мы используем t-критерий.

In [8]:
X = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [9]:
# Расчитываем вводные значения:
n = len(X)
M = Xcp = X.mean()
sigma = X.std(ddof=1)

t a/2 табличное находим в таблице для 95 доверительного интервала и степени свободы n-1:

In [10]:
t_a_2 = 2.262

In [11]:
lim1 = Xcp - t_a_2 * sigma/n**(1/2)
lim1

110.55660776308164

In [12]:
lim2 = Xcp + t_a_2 * sigma/n**(1/2)
lim2

125.64339223691834

##### Ответ: доверительный интервал для математического ожидания с надежностью 0.95 находится в интервале [110.55660776308164, 125.64339223691834]

##### 3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95. 

По условию задачи нам известна дисперсия, поэтому используем z-критерий.

In [13]:
# Дано:
Xcp = M = 174.2
n=27
sigma = 5

Находим Z табличное для значения 0.975 (берем для a/2):

In [14]:
Za_2 = 1.96

In [15]:
lim1 = Xcp - Za_2 * sigma/n**(1/2)
lim1

172.31398912064722

In [16]:
lim2 = M + Za_2 * sigma/n**(1/2)
lim2

176.08601087935276

##### Ответ: доверительный интервал для математического ожидания с надежностью 0.95 находится в интервале [172.31398912064722, 176.08601087935276]